WordPress/gutenberg
# Font Library: refactor client side install functions to work with revised API

## What?
This PR revises the client side installation to work with the new endpoints. 

## Why?
See #57688.

## How?
- Adds a new resolver for the `/font-families/<id>/font-faces` endpoint
- Implements a batch processing function to handle concurrent requests to the font-faces endpoint
- Splits `makeFormDataFromFontFamily` into two utilities: `makeFontFaceFormData` and `makeFontFamilyFormData`, to create form data in the format expected by the API

I think more refactoring can be done to improve readability and error handling, but I wanted to make few changes as possible to prove the concept.

**To-do:**
- Add tests
- Process the errors and set notices accordingly

## Testing Instructions
- Go to the Font Library and verify you can install a new font family

Note that installing font faces to an existing font family, or deleting, does not work yet because changes are needed to the REST Controller.

### Testing Instructions for Keyboard


## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/5375500/720a9017-1788-47f2-9266-29a2451a9329
